src/test/java/de/learnlib/ralib/oracles/mto/BranchMergingTest.java

```diff
@@ -0,0 +1,362 @@
+        SymbolicSuffix suffix2 = new SymbolicSuffix(suffixActions2);
+
+        PIV piv1 = new PIV();
+        piv1.put(p1, r1);
+        piv1.put(p2, r2);
+        Mapping<SymbolicDataValue, DataValue<?>> vals1 = new Mapping<SymbolicDataValue, DataValue<?>>();
+        vals1.put(r1, dv1);
+        vals1.put(r2, dv2);
+
+        SDT sdt1 = new SDT(Map.of(
+                new EqualityGuard(s1, r1), new SDT(Map.of(
+                        new SDTTrueGuard(s2), SDTLeaf.ACCEPTING)),
+                new EqualityGuard(s1, r2), new SDT(Map.of(
+                        new EqualityGuard(s2, r1), SDTLeaf.REJECTING,
+                        new DisequalityGuard(s2, r1), SDTLeaf.ACCEPTING)),
+                new SDTAndGuard(s1, new DisequalityGuard(s1, r1), new DisequalityGuard(s1, r2)), new SDT(Map.of(
+                        new SDTTrueGuard(s2), SDTLeaf.REJECTING))));
+
+        SDT sdt2 = new SDT(Map.of(
+                new EqualityGuard(s1, r1), SDTLeaf.ACCEPTING,
+                new EqualityGuard(s1, r2), SDTLeaf.ACCEPTING,
+				new SDTAndGuard(s1, new DisequalityGuard(s1, r1), new DisequalityGuard(s1, r2)), SDTLeaf.REJECTING));
+
+        oracle.changeSDT(sdt1, vals1, consts);
+        SymbolicDecisionTree actualSDT1 = mto.treeQuery(prefix1, suffix1).getSdt();
+        Assert.assertTrue(actualSDT1.isEquivalent(sdt1, new VarMapping()));
+
+        oracle.changeSDT(sdt2, vals1);
+        SymbolicDecisionTree actualSDT2 = mto.treeQuery(prefix2, suffix2).getSdt();
+        Assert.assertTrue(actualSDT2.isEquivalent(sdt2, new VarMapping()));
+    }
+
+    @Test
+    public void branchMergingIneqTest() {
+        final DataType D_TYPE = new DataType("double", BigDecimal.class);
+        final InputSymbol A = new InputSymbol("a", new DataType[] {D_TYPE});
+
+        SDTOracle oracle = new SDTOracle();
+        final Map<DataType, Theory> teachers = new LinkedHashMap<>();
+        DoubleInequalityTheory dit = new DoubleInequalityTheory(D_TYPE);
+        teachers.put(D_TYPE, dit);
+
+        Constants consts = new Constants();
+        JConstraintsConstraintSolver solver = TestUtil.getZ3Solver();
+        MultiTheoryTreeOracle mto = new MultiTheoryTreeOracle(oracle, teachers, consts, solver);
+
+        SuffixValueGenerator svgen = new SuffixValueGenerator();
+        ParameterGenerator pgen = new ParameterGenerator();
+        RegisterGenerator rgen = new RegisterGenerator();
+
+        SuffixValue s1 = svgen.next(D_TYPE);
+        SuffixValue s2 = svgen.next(D_TYPE);
+        Parameter p1 = pgen.next(D_TYPE);
+        Parameter p2 = pgen.next(D_TYPE);
+        Parameter p3 = pgen.next(D_TYPE);
+        Register r1 = rgen.next(D_TYPE);
+        Register r2 = rgen.next(D_TYPE);
+        Register r3 = rgen.next(D_TYPE);
+        DataValue dv1 = new DataValue(D_TYPE, BigDecimal.ONE);
+        DataValue dv2 = new DataValue(D_TYPE, BigDecimal.valueOf(2));
+        DataValue dv3 = new DataValue(D_TYPE, BigDecimal.valueOf(3));
+
+        // test 1
```

Comment:
It's a bit weird to have such a meaningless comment in this line. There is no "test 2" anywhere, and the current comment is not very informative on what this "test 1"  is about.